src/workers/hermes_worker.js

```diff
@@ -84,7 +90,11 @@ export default class HermesWorker extends PeriodicWorker {
   }
 
   async afterWorkLoop() {
-    await this.server.close();
-    await this.mongoClient.close();
+    try {
+      await this.workerTaskTrackingRepository.finishWork(this.workId);
```

Comment:
IMO: this is surplus. I wouldn't include the workId into the state of the worker. Instead I would introduce a local variable `workId` inside periodicWork, and fully handle it there.